web_switch_company_warning/static/src/js/switch_company_warning.js

```diff
@@ -33,7 +33,11 @@ openerp.web_switch_company_warning = function (instance) {
             });
 
             w.port.start();
-            w.port.postMessage(this.session.company_id);
+            w.port.postMessage(signature(this.session));
+
+            function signature(session) {
+                return [session.db, session.uid, session.company_id].join();
```

Comment:
Can you move this function outside of the init method to easily overwrite it on customer project module if needs?